java/src/main/java/com/google/cloud/dataproc/templates/gcs/GCStoBigTable.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (C) 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.dataproc.templates.gcs;
+
+import static com.google.cloud.dataproc.templates.util.TemplateConstants.*;
+
+import com.google.cloud.bigtable.data.v2.BigtableDataClient;
+import com.google.cloud.bigtable.data.v2.models.RowMutation;
+import com.google.cloud.dataproc.templates.BaseTemplate;
+import java.util.Iterator;
+import java.util.Objects;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.spark.api.java.function.ForeachPartitionFunction;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GCStoBigTable implements BaseTemplate, java.io.Serializable {
+
+  public static final Logger LOGGER = LoggerFactory.getLogger(GCStoBigTable.class);
+
+  private String projectID;
+  private String inputFileLocation;
+  private String bigTableInstanceId;
+  private String bigTableTableName;
+  private String bigTableProjectId;
+  private String bigTableColumnFamily;
+  private String inputFileFormat;
+
+  public GCStoBigTable() {
+
+    projectID = getProperties().getProperty(PROJECT_ID_PROP);
+    inputFileLocation = getProperties().getProperty(GCS_BT_INPUT_LOCATION);
+    bigTableInstanceId = getProperties().getProperty(GCS_BT_OUTPUT_INSTANCE_ID);
+    bigTableTableName = getProperties().getProperty(GCS_BT_OUTPUT_TABLE_NAME);
+    bigTableProjectId = getProperties().getProperty(GCS_BT_OUTPUT_PROJECT_ID);
+    bigTableColumnFamily = getProperties().getProperty(GCS_BT_OUTPUT_TABLE_COLUMN_FAMILY);
+    inputFileFormat = getProperties().getProperty(GCS_BT_INPUT_FORMAT);
+  }
+
+  @Override
+  public void runTemplate() {
+    validateInput();
+
+    SparkSession spark = null;
+    LOGGER.info("input format: {}", inputFileFormat);
+
+    try {
+      spark = SparkSession.builder().appName("GCS to Bigtable load").getOrCreate();
+
+      Dataset<Row> inputData = null;
+
+      switch (inputFileFormat) {
```

Comment:
Why do we need this switch-case?

Reply:
The template has the capability to process CSV, AVRO and Parquet files. The switch case is there in order to send the right parameters for the dataframe read operator. For instance, for CSV, we need to provide details on header and all but for parquet,avro there is no need to pass those values. That is the need for the switch case